libraries/chain_kv/include/b1/session/session.hpp

```diff
@@ -0,0 +1,1244 @@
+#pragma once
+
+#include <queue>
+#include <set>
+#include <stack>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+#include <variant>
+
+#include <b1/session/key_value.hpp>
+#include <b1/session/session_fwd_decl.hpp>
+#include <b1/session/shared_bytes.hpp>
+
+namespace eosio::session {
+
+// Defines a session for reading/write data to a cache and persistent data store.
+//
+// \tparam Persistent_data_store A type that persists data to disk or some other non memory backed storage device.
+// \tparam Cache_data_store A type that persists data in memory for quick access.
+// \code{.cpp}
+//  auto memory_pool = eosio::session::boost_memory_allocator::make();
+//  auto pds = eosio::session::make_rocks_data_store(nullptr, memory_pool);
+//  auto cds = eosio::session::make_cache(memory_pool);
+//  auto fork = eosio::session::make_session(std::move(pds), std::move(cds));
+//  {
+//      auto block = eosio::session::make_session(fork);
+//      {
+//          auto transaction = eosio::session::make_session(block);
+//          auto kv = eosio::session::make_kv("Hello", 5, "World", 5,
+//          transaction.backing_data_store()->memory_allocator()); transaction.write(std::move(kv));
+//      }
+//      {
+//          auto transaction = eosio::session::make_session(block);
+//          auto kv = eosio::session::make_kv("Hello", 5, "World2", 5,
+//          transaction.backing_data_store()->memory_allocator()); transaction.write(std::move(kv));
+//      }
+//  }
+// \endcode
+template <typename Persistent_data_store, typename Cache_data_store>
```

Comment:
This shouldn't necessarily be a `persistent` data store, it is just it's parent type, that parent type might be something that is more persistent, but this type shouldn't make any assumptions about that.